### PR TITLE
#8 implement application layout using bootstrap

### PR DESCRIPTION
**Pull request by Sontsevolosa**
* Customization for using of Bootstrap
* Adding header and CSS for it
* Adding footer and CSS for it
* Writing simple tests
* Customization for testing output

@mluts please, review my code

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,8 +13,8 @@ rerun.txt
 pickle-email-*.html
 
 # TODO Comment out these rules if you are OK with secrets being uploaded to the repo
-config/initializers/secret_token.rb
-config/secrets.yml
+# config/initializers/secret_token.rb
```
Comment (mluts) [inline .gitignore]: why did you comment these 2 lines ?

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,92 @@
+@import "bootstrap-sprockets";
+@import "bootstrap";
+
+/* universal */
+
+body {
+	padding-top: 60px;
```
Comment (mluts) [inline app/assets/stylesheets/custom.scss]: @Sontsevolosa tabs detected, please don't use tabs here

File: app/assets/stylesheets/lunches.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Lunches controller here.
```
Comment (mluts) [inline app/assets/stylesheets/lunches.scss]: Unnecessary scss file

File: app/assets/javascripts/lunches.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (mluts) [inline app/assets/javascripts/lunches.coffee]: Unnecessary coffee file

File: app/controllers/lunches_controller.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class LunchesController < ApplicationController
```
Comment (mluts) [inline app/controllers/lunches_controller.rb]: @Sontsevolosa Why did you add this controller ? There are no tasks to do this

File: app/helpers/lunches_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module LunchesHelper
```
Comment (mluts) [inline app/helpers/lunches_helper.rb]: Unused module

File: app/views/lunches/index.html.slim [hunk 7]
```diff
@@ -0,0 +1 @@
+h1 Hello everyone!
```
Comment (mluts) [inline app/views/lunches/index.html.slim]: Remove this page please

File: app/views/layouts/_footer.html.slim [hunk 8]
```diff
@@ -0,0 +1,9 @@
+footer
```
Comment (mluts) [inline app/views/layouts/_footer.html.slim]: @Sontsevolosa footer and header are OK, they are part of global layout. But lunches controller and files related to it are unnecessary

File: .gitignore (hunk 1 above)
Comment (Sontsevolosa) [inline .gitignore]: for CircleCI. it needed information from secrets.yml which was ignored by git

File: app/views/layouts/application.html.slim [hunk 9]
```diff
@@ -1,10 +1,28 @@
...
+    
     title Lunchez
     = stylesheet_link_tag 'application', media: 'all'
     = javascript_include_tag 'application'
     = csrf_meta_tags
+
+    link href="css/bootstrap.min.css" rel="stylesheet"
+    link href="custom.scss" rel="stylesheet"
+
+    /[if lt IE 9]
```
Comment (mluts) [inline app/views/layouts/application.html.slim]: Why did you add this section ? What it does?

File: app/views/layouts/application.html.slim [hunk 10]
```diff
@@ -1,10 +1,28 @@
 doctype html
 html
   head
+    meta charset="utf-8"
+    meta http-equiv="X-UA-Compatible" content="IE=edge"
+    meta name="viewport" content="width=device-width, initial-scale=1"
```
Comment (mluts) [inline app/views/layouts/application.html.slim]: Could you explain me what it does? I'm not sure

File: app/views/layouts/application.html.slim [hunk 11]
```diff
@@ -1,10 +1,28 @@
 doctype html
 html
   head
+    meta charset="utf-8"
+    meta http-equiv="X-UA-Compatible" content="IE=edge"
```
Comment (mluts) [inline app/views/layouts/application.html.slim]: Is this for IE compatability ?

File: app/controllers/lunches_controller.rb (hunk 5 above)
Comment (Sontsevolosa) [inline app/controllers/lunches_controller.rb]: for verify my layout

File: .gitignore (hunk 1 above)
Comment (mluts) [inline .gitignore]: @Sontsevolosa please revert changes in `.gitignore`, and add preparation command to CircleCI. 

Instead of commenting gitignore, you can add `cp config/secrets.yml.example config/secrets.yml` to `circle.yml` in some section, and it should work

File: app/controllers/lunches_controller.rb (hunk 5 above)
Comment (mluts) [inline app/controllers/lunches_controller.rb]: @Sontsevolosa but this is out of the scope of current task

Comment (mluts) [inline app/controllers/lunches_controller.rb]: @Sontsevolosa okay, let's leave it as it is

File: app/views/lunches/index.html.slim (hunk 7 above)
Comment (mluts) [inline app/views/lunches/index.html.slim]: @Sontsevolosa sorry, you can not removing this page. Decided to leave this page, as there are really no pages to "touch" application.

File: app/views/layouts/_footer.html.slim (hunk 8 above)
Comment (mluts) [inline app/views/layouts/_footer.html.slim]: @Sontsevolosa ok with lunches controller

**Comment by mluts** [comment]
@Sontsevolosa Please consider my comments

File: app/assets/stylesheets/custom.css [hunk 12]
```diff
@@ -0,0 +1,92 @@
+@import "bootstrap-sprockets";
```
Comment (mluts) [inline app/assets/stylesheets/custom.css]: @Sontsevolosa Please use `.scss` extension

Comment (mluts) [inline app/assets/stylesheets/custom.css]: @Sontsevolosa why did you use `custom.css` and did not add these changes to `application.css` ?

File: app/views/layouts/application.html.slim (hunk 10 above)
Comment (Sontsevolosa) [inline app/views/layouts/application.html.slim]: this is configuration for bootstrap. I took this from [this](http://getbootstrap.com/getting-started/#template)

File: app/assets/stylesheets/custom.css [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+@import "bootstrap";
+
+/* universal */
```
Comment (mluts) [inline app/assets/stylesheets/custom.css]: @Sontsevolosa Or, as i see that you have divided `custom.css` by sections. These sections can live in separate files like `universal.scss`, `typography.scss` and so on. This is on your choice - to do this or what i described above

File: app/views/layouts/application.html.slim (hunk 10 above)
Comment (mluts) [inline app/views/layouts/application.html.slim]: @Sontsevolosa ok, it is enough. Could you leave short explanation comment in this code, that you have took it from bootstrap documentation ? I think it is not obvious for most people.

File: app/views/layouts/application.html.slim (hunk 9 above)
Comment (Sontsevolosa) [inline app/views/layouts/application.html.slim]: because it was in bootstrap documentation

**Comment by Sontsevolosa** [comment]
@mluts please, review my code

File: app/assets/stylesheets/custom.scss [hunk 14]
```diff
@@ -0,0 +1,92 @@
+@import "bootstrap-sprockets";
```
Comment (mluts) [inline app/assets/stylesheets/custom.scss]: @Sontsevolosa please do one of the following:

* Split this file according to sections in this file (e.g. `universal.scss`, `typography.scss`)
* Or move this code to `application.scss`

`custom.scss` has general meaning, same as `application.scss`. I think it is no good to create unnecessary general-purpose files.

File: app/views/layouts/application.html.slim [hunk 15]
```diff
@@ -1,10 +1,29 @@
 doctype html
 html
   head
+    / This is code from Bootstrap documentation
```
Comment (mluts) [inline app/views/layouts/application.html.slim]: @Sontsevolosa could you attach link to bootstrap docs here ?

File: config/routes.rb [hunk 16]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  root 'lunches#index'
+  get 'lunches/index'
```
Comment (mluts) [inline config/routes.rb]: @Sontsevolosa why did you add `get` here? I think `root` is enough...

File: config/secrets.yml [hunk 17]
```diff
@@ -0,0 +1,22 @@
+# Be sure to restart your server when you modify this file.
```
Comment (mluts) [inline config/secrets.yml]: Please don't commit secrets.yml

File: app/views/layouts/application.html.slim (hunk 15 above)
Comment (Sontsevolosa) [inline app/views/layouts/application.html.slim]: @mluts sorry, I changed my mind.
I implemented Bootstrap using normal CDN.
Early I did it wrong...

File: app/views/layouts/application.html.slim [hunk 18]
```diff
@@ -5,6 +5,19 @@ html
...
     = csrf_meta_tags
+
+    link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous"
+
+    /[if lt IE 9]
+      javascript:
+      src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"
+      src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"
+    
+    = render 'layouts/header'
```
Comment (mluts) [inline app/views/layouts/application.html.slim]: @Sontsevolosa mistake here, you render header in `head` tag

File: config/secrets1.yml [hunk 19]
```diff
@@ -0,0 +1,22 @@
+# Be sure to restart your server when you modify this file.
+
```
Comment (mluts) [inline config/secrets1.yml]: @Sontsevolosa Don't commit this file

**Comment by mluts** [comment]
@Sontsevolosa please consider my comments

**Comment by Sontsevolosa** [comment]
@mluts please, review my changes

File: app/assets/stylesheets/application.scss [hunk 20]
```diff
@@ -0,0 +1,108 @@
...
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any styles
+ * defined in the other CSS/SCSS files in this directory. It is generally better to create a new
+ * file per style scope.
+ *
+ *= require_tree .
+ *= require_self
+ */
+
+ @import "bootstrap-sprockets";
```
Comment (mluts) [inline app/assets/stylesheets/application.scss]: @Sontsevolosa remove redundant space here

File: app/controllers/lunches_controller.rb [hunk 21]
```diff
@@ -0,0 +1,4 @@
...
+  def index
```
Comment (mluts) [inline app/controllers/lunches_controller.rb]: @Sontsevolosa good practice is at least add `render` here, because in fact it is implicitly called here

File: app/assets/stylesheets/application.scss [hunk 22]
```diff
@@ -0,0 +1,108 @@
...
+@import "bootstrap";
```
Comment (mluts) [inline app/assets/stylesheets/application.scss]: @Sontsevolosa you have already one bootstrap taken from CDN in application layout. Remove this bootstrap from `application.scss` and corresponding gem from Gemfile

**Comment by mluts** [comment]
@Sontsevolosa Have only few questions, please review my comments

**Comment by Sontsevolosa** [comment]
@mluts please, review

**Comment by Sontsevolosa** [comment]
@mluts can I make merge?

**Comment by mluts** [comment]
@Sontsevolosa Sorry for the delay, reviewing

File: app/views/layouts/application.html.slim [hunk 23]
```diff
@@ -5,6 +5,20 @@ html
...
     = csrf_meta_tags
+
+    link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous"
+
+    /[if lt IE 9]
+      javascript:
+      src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"
+      src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"
+    
+
```
Comment (mluts) [inline app/views/layouts/application.html.slim]: @Sontsevolosa Unnecessary 2 newlines

**Comment by mluts** [comment]
@Sontsevolosa please remove unnecessary newlines and merge

File: app/views/layouts/application.html.slim (hunk 23 above)
Comment (Sontsevolosa) [inline app/views/layouts/application.html.slim]: @mluts  done
